core/src/test/java/kafka/admin/ConfigCommandIntegrationTest.java

```diff
@@ -65,219 +75,494 @@
+        alterAndVerifyConfig(zkClient, adminZkClient, Optional.empty(),
+                singletonMap("message.max.bytes", "140000"));
 
         // Delete config
-        deleteAndVerifyConfig(zkClient, Collections.singleton("message.max.size"), Optional.of(brokerId));
-        deleteAndVerifyConfig(zkClient, Collections.singleton("message.max.size"), Optional.empty());
+        deleteAndVerifyConfig(zkClient, adminZkClient, Optional.of(brokerId),
+                singleton("message.max.bytes"));
+        deleteAndVerifyConfig(zkClient, adminZkClient, Optional.empty(),
+                singleton("message.max.bytes"));
 
         // Listener configs: should work only with listener name
-        alterAndVerifyConfig(zkClient, Collections.singletonMap("listener.name.external.ssl.keystore.location", "/tmp/test.jks"), Optional.of(brokerId));
+        alterAndVerifyConfig(zkClient, adminZkClient, Optional.of(brokerId),
+                singletonMap("listener.name.internal.ssl.keystore.location", "/tmp/test.jks"));
         assertThrows(ConfigException.class,
-            () -> alterConfigWithZk(zkClient, Collections.singletonMap("ssl.keystore.location", "/tmp/test.jks"), Optional.of(brokerId)));
+                () -> alterConfigWithZk(zkClient, adminZkClient, Optional.of(brokerId),
+                        singletonMap("ssl.keystore.location", "/tmp/test.jks")));
 
         // Per-broker config configured at default cluster-level should fail
         assertThrows(ConfigException.class,
-            () -> alterConfigWithZk(zkClient, Collections.singletonMap("listener.name.external.ssl.keystore.location", "/tmp/test.jks"), Optional.empty()));
-        deleteAndVerifyConfig(zkClient, Collections.singleton("listener.name.external.ssl.keystore.location"), Optional.of(brokerId));
+                () -> alterConfigWithZk(zkClient, adminZkClient, Optional.empty(),
+                        singletonMap("listener.name.internal.ssl.keystore.location", "/tmp/test.jks")));
+        deleteAndVerifyConfig(zkClient, adminZkClient, Optional.of(brokerId),
+                singleton("listener.name.internal.ssl.keystore.location"));
 
         // Password config update without encoder secret should fail
         assertThrows(IllegalArgumentException.class,
-            () -> alterConfigWithZk(zkClient, Collections.singletonMap("listener.name.external.ssl.keystore.password", "secret"), Optional.of(brokerId)));
+                () -> alterConfigWithZk(zkClient, adminZkClient, Optional.of(brokerId),
+                        singletonMap("listener.name.internal.ssl.keystore.password", "secret")));
 
         // Password config update with encoder secret should succeed and encoded password must be stored in ZK
         Map<String, String> configs = new HashMap<>();
-        configs.put("listener.name.external.ssl.keystore.password", "secret");
+        configs.put("listener.name.internal.ssl.keystore.password", "secret");
```

Comment:
why we need this change?

Reply:
`external` means there is a listener called `external` and we want to set its property `ssl.keystore.password`.

Reply:
I was wrong. Appreciate your feedback.